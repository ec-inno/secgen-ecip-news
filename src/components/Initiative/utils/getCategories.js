const getCategories = t => {
  return [
    { value: 'any', label: t('All categories') },
    { value: 'AGRI', label: t('Agriculture') },
    { value: 'DEVCO', label: t('Aid and development cooperation') },
    { value: 'EURO', label: t('Business and economy') },
    { value: 'SANTE', label: t('Consumers and health') },
    { value: 'CULT', label: t('Culture and media') },
    { value: 'DECO', label: t('Digital economy and society') },
    { value: 'EDU', label: t('Education, youth and sport') },
    { value: 'EMPL', label: t('Employment and social affairs') },
    { value: 'ENER', label: t('Energy') },
    { value: 'ENV', label: t('Environment and climate') },
    { value: 'TRADE', label: t('External trade and relations') },
    { value: 'JUST', label: t('Justice and fundamental rights') },
    { value: 'MARE', label: t('Maritime affairs and fisheries') },
    { value: 'MIGR', label: t('Migration and asylum') },
    { value: 'REGIO', label: t('Regional development') },
    { value: 'RSH', label: t('Research') },
    { value: 'SEC', label: t('Security') },
    { value: 'TRA', label: t('Transport') },
    { value: 'REGFRA', label: t('REGFRA') },
    { value: 'REGECI', label: t('REGECI') },
    { value: 'COLLECI', label: t('COLLECI') },
    { value: 'EXAECI', label: t('EXAECI') },
  ];
};

export default getCategories;
