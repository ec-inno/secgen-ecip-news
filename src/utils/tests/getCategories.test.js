import getCategories from '../getCategories';

describe('getCategories utlity', () => {
  it('returns a list of values', () => {
    const t = s => s; // mocks https://www.i18next.com/overview/api#t
    const categories = getCategories(t);
    expect(categories).toMatchSnapshot();
  });
});
