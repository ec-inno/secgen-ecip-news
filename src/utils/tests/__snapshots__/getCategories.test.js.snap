// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCategories utlity returns a list of values 1`] = `
Array [
  Object {
    "label": "All categories",
    "value": "any",
  },
  Object {
    "label": "Agriculture",
    "value": "AGRI",
  },
  Object {
    "label": "Aid and development cooperation",
    "value": "DEVCO",
  },
  Object {
    "label": "Business and economy",
    "value": "EURO",
  },
  Object {
    "label": "Consumers and health",
    "value": "SANTE",
  },
  Object {
    "label": "Culture and media",
    "value": "CULT",
  },
  Object {
    "label": "Digital economy and society",
    "value": "DECO",
  },
  Object {
    "label": "Education, youth and sport",
    "value": "EDU",
  },
  Object {
    "label": "Employment and social affairs",
    "value": "EMPL",
  },
  Object {
    "label": "Energy",
    "value": "ENER",
  },
  Object {
    "label": "Environment and climate",
    "value": "ENV",
  },
  Object {
    "label": "External trade and relations",
    "value": "TRADE",
  },
  Object {
    "label": "Justice and fundamental rights",
    "value": "JUST",
  },
  Object {
    "label": "Maritime affairs and fisheries",
    "value": "MARE",
  },
  Object {
    "label": "Migration and asylum",
    "value": "MIGR",
  },
  Object {
    "label": "Regional development",
    "value": "REGIO",
  },
  Object {
    "label": "Research",
    "value": "RSH",
  },
  Object {
    "label": "Security",
    "value": "SEC",
  },
  Object {
    "label": "Transport",
    "value": "TRA",
  },
  Object {
    "label": "Regulatory Framework",
    "value": "REGFRA",
  },
  Object {
    "label": "Registration of initiatives",
    "value": "REGECI",
  },
  Object {
    "label": "Signature collection",
    "value": "COLLECI",
  },
  Object {
    "label": "Examination of initiatives",
    "value": "EXAECI",
  },
]
`;
